### PR TITLE
Make supervision strategy configurable.

### PR DESCRIPTION
**Pull request by bjhaid**
Related issue/PRs:

https://github.com/kafkaex/kafka_ex/pull/92
https://github.com/kafkaex/kafka_ex/issues/65

@dantswain can you please review

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -17,4 +17,13 @@ config :kafka_ex,
...
-  sync_timeout: 3000
+  sync_timeout: 3000,
+  # Supervision Strategy
+  # Modify this if you want to configure the strategy the supervisor will use
+  # to restart the workers.
+  # if you need help configuring the supervision strategy see:
+  # http://elixir-lang.org/docs/stable/elixir/Supervisor.Spec.html#supervise/2
+  # The default supervision strategy is as configured below:
+  supervision_strategy: [
+    strategy: :simple_one_for_one, max_restarts: 10, max_seconds: 60
```
Comment (dantswain) [inline config/config.exs]: I think you just want to expose `max_restarts` and `max_seconds`.  The strategy itself pretty much has to be `:simple_one_for_one` for `KafkaEx.create_worker` to work.

Comment (bjhaid) [inline config/config.exs]: Thanks you are correct!

**Comment by bjhaid** [comment]
@dantswain how about now? What do you think about being added as a collaborator on kafka_ex (i.e getting commit access to the repo)

File: test/supervisor_test.exs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  use ExUnit.Case, async: false
+
+  test "uses configured max_restarts" do
+    # stop the application, configure supervision max_restarts and restart the application
+    configured_max_restarts = 15
+
+    :ok = Application.stop(:kafka_ex)
+    Application.put_env(:kafka_ex, :max_restarts, configured_max_restarts)
+    {:ok, _} = Application.ensure_all_started(:kafka_ex)
+    [max_restarts, _max_seconds] = :sys.get_state(KafkaEx.Supervisor) |> elem(9)
```
Comment (dantswain) [inline test/supervisor_test.exs]: Using `elem` to dig into the supervisor's state is prone to breaking if the structure changes.  Being just test code it's not TOO bad, but maybe better embedded in a function?  I looked at the supervisor source code and don't see any way to introspect the state and had no luck :/

TBH I'd be comfortable not testing this.  I'm a big believer in testing, but this seems like a lot of work to test something pretty straightforward.

**Comment by dantswain** [comment]
I think the implementation is good.  I left a comment on the tests.

I'm fine with commit access, but I'll leave that up to you.  I'm honored by the offer.

File: test/supervisor_test.exs (hunk 2 above)
Comment (bjhaid) [inline test/supervisor_test.exs]: I thought alot about this test, I am using an undocumented API in the test, but I convinced myself it's necessary to guide future refactoring (i.e. some spec that will break if something is wrongly modified)

Comment (dantswain) [inline test/supervisor_test.exs]: Heh.  This is failing for me with Erlang R18.3 and Elixir 1.2.3.  This is the state I get back:

```
{:state, {:local, KafkaEx.Supervisor}, :simple_one_for_one, [{:child, :undefined, KafkaEx.Server, {KafkaEx.Server, :start_link, []}, :permanent, 5000, :worker, [KafkaEx.Server]}], {:dict, 1, 16, 16, 8, 80, 48, {[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []}, {{[], [[#PID<0.264.0>, [uris: [{"localhost", 9092}, {"localhost", 9093}, {"localhost", 9094}], consumer_group: "kafka_ex"], KafkaEx.Server]], [], [], [], [], [], [], [], [], [], [], [], [], [], []}}}, 15, 120, [], 0, KafkaEx.Supervisor, [15, 120]}
```

Comment (bjhaid) [inline test/supervisor_test.exs]: I will delete this spec

**Pull request merged**